scripts/pylib/twister/twisterlib/testplan.py

```diff
@@ -548,6 +646,72 @@ def add_testsuites(self, testsuite_filter=None):
                                 suite.platform_allow,
                                 f"platform_allow in {suite.name}")
 
+                        if self.options.check:
+                            checker = self.checker
+                            if len(suite.platform_allow) > MAX_PLATFORM_ALLOW:
+                                checker.add(
+                                    checker.ALLOW,
+                                    f"Too many entries in allow_platform in {suite.name}: "
+                                    f"{len(suite.platform_allow)}"
+                                )
+
+                            if len(suite.integration_platforms) > MAX_INTEGRATION_PLATFORMS:
+                                checker.add(
+                                    checker.INTEGRATION,
+                                    f"Too many integration platforms in {suite.name}: "
+                                    f"{len(suite.integration_platforms)}"
+                                )
+
+                            if len(suite.platform_exclude) > MAX_PLATFORM_EXCLUDE:
+                                checker.add(
+                                    checker.EXCLUDE,
+                                    f"Too many excluded platforms in {suite.name}: "
+                                    f"{len(suite.platform_exclude)}"
+                                )
+
+                            if len(suite.platform_allow) > 1 and not suite.integration_platforms:
+                                checker.add(
+                                    checker.ALLOW_NO_INTEGRATION,
+                                    f"platform_allow in {suite.name} without integration_platforms"
+                                )
+
+                            if suite.build_only and "build" not in suite.id:
+                                checker.add(checker.BUILD_ONLY,f"build_only set in {suite.name}")
+
+                            if suite.skip:
+                                checker.add(checker.SKIP,f"skip set in {suite.name}")
+
+                            if suite.harness not in SUPPORTED_HARNESS:
+                                checker.add(
+                                    checker.HARNESS,
+                                    f"Unsupported harness {suite.harness} in {suite.name}"
+                                )
+
+                            if (suite.filter and not suite.integration_platforms and
+                                not suite.platform_allow and not suite.platform_key):
+                                checker.add(
+                                    checker.FILTER_NO_INTEGRATION,
+                                    f"filter with no integration platforms in {suite.name}"
```

Comment:
the reason string doesn't correspond to the logic (not mentioning platform_key and platform_allow)